src/boot.c

```diff
@@ -412,31 +427,152 @@ static void read_fsinfo(DOS_FS * fs, struct boot_sector *b, unsigned int lss)
 	fs->free_clusters = le32toh(i.free_clusters);
 }
 
+static int is_fat32(const struct boot_sector *b)
+{
+    return !b->fat_length && b->fat32_length;
+}
+
+static int try_boot_sector(unsigned *boot_sector, unsigned offset,
+                           struct boot_sector *b)
+{
+    unsigned int logical_sector_size;
+    off_t byte_offset = offset * SECTOR_SIZE;
+
+    fs_read(byte_offset, sizeof(*b), b);
+    logical_sector_size = GET_UNALIGNED_W(b->sector_size);
+
+    if (
+	!logical_sector_size ||
+	/* Logical sector size is zero. */
+
+	/* This is the first thing that will fail if the platform needs special
+	 * handling of unaligned multibyte accesses but such handling isn't
+	 * being provided. See GET_UNALIGNED_W() above. */
+	logical_sector_size & (SECTOR_SIZE - 1) ||
+	/* Logical sector size is not a multiple of the physical sector size. */
+
+	!b->cluster_size ||
+	/* Cluster size is zero. */
+
+	!b->fats ||
+	/* Number of FATs is zero. (NTFS is like this.) */
+
+	byte_offset % logical_sector_size
+	/* ...else the boot sector isn't on a logical sector boundary. */
+	)
```

Comment:
As written in another comment https://github.com/dosfstools/dosfstools/pull/86#discussion_r692547126, this check is incomplete. It allows to use e.g. backup boot sector which is outside of the reserved area space, which is obviously invalid and would not work. So proper check for fat boot sector needs to be done.